content/Tjenester/klientpakker.md

```diff
@@ -0,0 +1,39 @@
+---
+title: Distribuerte pakker og klienter - risikovurdering
+date: 2023-09-19
+alias: [/fiks-plattform/tjenester/distribuertepakker/]
+---
+
+## Kort beskrivelse
+
+KS leverer pakker for Java og .NET gjennom henholdsvis Maven Central og Nuget.org for flere av tjenestene våre.
+Eskempelvis Fiks-IO klient, Maskinporten klient og pakker til protokollene Fiks-Arkiv, Fiks-Plan osv. 
+
+
+## Risikovurderinger og tiltak
+
+Det er gjort vurderinger og tiltak for å sikre pakkene som leveres.
+Risiko kan ligge i kompromitterte pakker i de sentraliserte repositories. 
+Dette kan hende som resultat av:
+- Tredjepart laster opp kompromitterte pakker til Maven Central eller Nuget.org som KS
+- At man har et kompromittert byggemiljø som leverer kompromitterte pakker
+- Avhengige pakker, tredjepart pakker, som følger med KS sine pakker er komprimettert og følger med ut med KS sine pakker
+
+
+### Maven
+
+_Kommer_
```

Comment:
- Alle pakker som skal distribueres til sentral maven distribusjonstjener signeres med egen GPG nøkkel eid av KS
- Den private delen av GPG nøkkelen og passkey er lagret som credential på bygg server og er bare tilgjengelig for bygg-agenten når den skal publisere pakken
- Pakkene lastes opp til egen konto på sentral distribusjontjeneste som har tilgang til namespace/groupId `no.ks`. Ingen andre kontoer vil kunne distribuere pakker under dette namespace (og underliggende namespace)
- Sentral distribusjonstjener gjør automatiserte tester på pakken for å sjekke at den følger formelle krav før den blir gjort tilgjengelig

Reply:
Burde kanskje også skrevet at passord til kontoen vår på sentral distribusjonsserver er lagret i en credential store på byggserver